### PR TITLE
Use process.stdout.write to print the report

### LINKED ISSUE #35: What is wrong with v0.2.0?
Hi, I just upgraded from 0.1.5 to 0.2.0... and my console blows up. Total mess, lots of CSS code instead of errors and warnings, all without any formating, just plain text, totaly unreadable.

I'm not using any special formaters, just bulk run without any parameters.

Also on some files I get this error
```
         output += reporter.formatResults(file.csslint.originalReport, file.p
                  ^
RangeError: Invalid string length
```
PS. I'm not only one, my teammates have same problems.

### PR DESCRIPTION
**Pull request by SimenB**
Fixes #35 

Testing it became quite hard, so ended up introducing logger option, which is provided a sinon mock from the test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,6 +102,28 @@ gulp.task('lint', function() {
...
+See the documentation for the formatters regarding what options they support.
+
+This plugin supports one option outside of that, called `logger`, allowing you to specify how to log out the report.
+Default is using `process.stdout.write`, but you can use e.g. `console.log`.
+
+```js
+gulp.task('lint', function() {
+  gulp.files('lib/*.css')
+    .pipe(csslint())
+    .pipe(csslint.reporter('junit-xml', {logger: console.log.bind(console.log)}));
```
Comment (lazd) [inline README.md]: Shouldn't that be `console.log.bind(console)`?

**Comment by lazd** [comment]
I'm not sure about writing to `stdout` directly, I think we should use [gulp-util's `log()` method](https://github.com/gulpjs/gulp-util#logmsg) in order to be a good citizen...

Also, I wonder if we need to expose an option at all for the logger. For now, wouldn't it just make sense to call `gutil.log()`?

**Comment by SimenB** [comment]
`gutil.log` is just `console.log` with the time before it, isn't it? The whole point here is to not use `console.log`, as it adds a line break after each invocation. And joining the string apparently explodes if it's long enough

File: README.md (hunk 1 above)
Comment (SimenB) [inline README.md]: You are absolutely right!

**Comment by lazd** [comment]
You're right, `process.stdout` must be the right thing to do. Giving an option for the logger lets folks do whatever they want with the output, so they can avoid printing to `stdout` if that's undesirable.

**Comment by SimenB** [comment]
I could maybe use `gutil.log` in the example in the readme instead of console.log?

Or we could go the other way around, and have a big message saying "if you get `RangeError: Invalid string length`, use `process.stdout`"

**Comment by lazd** [comment]
Yeah, showing `gutil.log` might be nice, as the `[gulp]` prefix and whatnot may be expected by some users. On the other hand, it may interfere with the report output. It's up to you on this one.

**Comment by SimenB** [comment]
@lazd How about now.

I also added an example if you want to print to file. I think this satisfies #34, and this I'm closing it if this is acceptable.

File: README.md [hunk 2]
```diff
@@ -102,6 +102,56 @@ gulp.task('lint', function() {
...
+
+  gulp.files('lib/*.css')
+    .pipe(csslint())
+    .pipe(csslint.reporter('junit-xml', {logger: function(str) { output += str; }}));
+
+  fs.writeFile('some/path/junit.xml', output, cb);
+});
+```
+
+This functionality is only available when not using custom reporters.
```
Comment (SimenB) [inline README.md]: This small note could be removed when #29 comes in, dropping support for custom

**Comment by SimenB** [comment]
@lazd Thoughts?

**Comment by lazd** [comment]
Let's merge this now, then quickly get #29 together, then do a release. That way all the reporter stuff happens at the same time. Sounds right, @SimenB?

**Comment by SimenB** [comment]
Aye, sounds good

**Comment by SimenB** [comment]
@lazd Rebased this

**Comment by lazd** [comment]
@SimenB if we're good to go here, I can merge this. Do you want to do #29 immediately after and push a release or should we wait on that?

**Comment by SimenB** [comment]
This is done, yes.

Regarding @29 I'd like to get it done, but I'm quite busy the next couple of weeks, and won't be able to wrap it up until after that

**Comment by lazd** [comment]
Ok, I'll go ahead and merge/release this as a patch release.

**Comment by 29** [comment]
Huh?

**Comment by lazd** [comment]
@29 sorry, I believe we meant #29 :)

**Comment by SimenB** [comment]
Oh yeah haha. Sorry about that!

**Comment by SimenB** [comment]
@lazd ping :smile:

**Comment by lazd** [comment]
@SimenB sorry for the delay, merging now!

**Comment by lazd** [comment]
Released in 0.3.0!

**Comment by SimenB** [comment]
Sweet! I'll try to wrap up #29 ASAP then, and a 1.0 release should be GTG :smile:

**Comment by lazd** [comment]
@SimenB awesome!

**Pull request merged**